### PR TITLE
feat: add supports of filtering and sorting to iot-table

### PR DESCRIPTION
**Pull request by square-li**
## Overview
Release Table component V2. Adding supports for filtering and sorting.

## Tests
[Include a link to the passing GitHub action running the test suite here.]

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: docs/Table.md [hunk 1]
```diff
@@ -19,41 +19,68 @@ const { query } = initialize({ iotsitewiseClient });
...
+  items={[
     {
-      header: 'Tempature',
-      rows: [
-        toId({ assetId: 'engine-1', propertyId: 'tempature' }),
-        toId({ assetId: 'engine-2', propertyId: 'tempature' }),
-      ]
+      rpm: {
+        $cellRef: {
+          id: toId({ assetId: 'turbine-1', propertyId: 'rotation_per_minute' }),
```
Comment (diehbria) [inline docs/Table.md:26]: add import statement for `toId`

File: docs/Table.md [hunk 2]
```diff
@@ -72,86 +99,192 @@ const { query } = initialize({ iotsitewiseClient });
...
-Specifies the row and column structure for the table, provides a mechanism to map the time series data queried for, into the cells of your choice
-Type: Array
+### `items`
+Specifies the data displayed in rows. Each item contains the data for one row. The display of a row is handled by the cell property of each column definition in the columnDefinitions property.
 
-Each table column contains the following properties
+Type: Array of [Item](#item) objects
 
-`header`
+Required: Yes
```
Comment (diehbria) [inline docs/Table.md:178]: The style is to only specify if optional, by stating `(Optional)` at the start of the paragraph. No need to state if required

File: docs/Table.md [hunk 3]
```diff
@@ -72,86 +99,192 @@ const { query } = initialize({ iotsitewiseClient });
...
+        resolution: 0
+      }
+    },
+  }
+] 
+```
+- `resolution`
+The resolution, in milliseconds, at which the data should be aggregated.
+  - Type: Number  
+  - Required: Yes
```
Comment (diehbria) [inline docs/Table.md:235]: remove

File: packages/table/src/utils/createCellItem.spec.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { createCellItem } from './createCellItem';
+import { CellItem } from './types';
+
+describe('createCellItem', () => {
```
Comment (diehbria) [inline packages/table/src/utils/createCellItem.spec.ts:4]: in context of `createCellItem` so don't need extra file wide describe block

File: packages/table/src/utils/createCellItem.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  isLoading?: boolean;
+  threshold?: Threshold;
+};
+export const createCellItem: (props?: CellProps) => CellItem = ({ value, error, isLoading, threshold } = {}) => {
+  const valueOf = () => {
+    if (error) {
+      return error.msg;
+    }
+    if (isLoading) {
+      return 'Loading';
```
Comment (diehbria) [inline packages/table/src/utils/createCellItem.ts:17]: this should be customizable via `messageOverrides`

File: packages/table/src/utils/tableHelpers.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+import { LoadingSpinner } from './spinner';
+import './style.css';
+
+export const getDefaultColumnDefinitions: (
+  columnDefinitions: ColumnDefinition[]
+) => (AWSUITableProps.ColumnDefinition<TableItem> & ColumnDefinition)[] = (columnDefinitions) => {
+  return columnDefinitions.map((colDef) => ({
+    cell: (item: TableItem) => {
+      if (!(colDef.key in item)) {
+        return <div>-</div>;
```
Comment (diehbria) [inline packages/table/src/utils/tableHelpers.tsx:16]: does this need to return a div? Could this just return a fragment `<>-</>` instead?

File: packages/table/src/table/index.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React, { FunctionComponent } from 'react';
+import { PropertyFilter, Table as AWSUITable } from '@awsui/components-react';
+import { useCollection } from '@awsui/collection-hooks';
+import { TableProps } from '../utils';
+import { defaultI18nStrings, getDefaultColumnDefinitions } from '../utils/tableHelpers';
```
Comment (diehbria) [inline packages/table/src/table/index.tsx:5]: no way to pass these in (via `messageOverrides`)

**Review by diehbria — CHANGES_REQUESTED** [review]

**Comment by diehbria** [comment]
Should this be a major version change with documented changes in the change logs?

**Comment by diehbria** [comment]
are there any tests related to displaying non-latest data when there are cached more recent data (i.e. that the cells don't simply always display the latest data point, as KPI/Grid/Old table did

**Comment by tracfren** [comment]
What are the next steps to merge this PR in?

**Comment by square-li** [comment]
Adding supports of messageOverrrides by this PR: https://github.com/awslabs/iot-app-kit/pull/211.
Other requested changes will be added shortly as separated PRs also.

**Comment by square-li** [comment]
Updating docs and bump version by this PR: https://github.com/awslabs/iot-app-kit/pull/213

File: packages/table/src/utils/tableHelpers.tsx (hunk 6 above)
Comment (square-li) [reply packages/table/src/utils/tableHelpers.tsx:16]: Removed.

**Comment by square-li** [comment]
Closing this PR for cleaning up git history. 
New PR: https://github.com/awslabs/iot-app-kit/pull/221

**Pull request closed without merging**